compiler/src/dotty/tools/dotc/transform/LazyVals.scala

```diff
@@ -274,6 +274,231 @@ class LazyVals extends MiniPhase with IdentityDenotTransformer {
     }
   }
 
+  /**
+   * Create a threadsafe lazy accessor and function that computes the field's value. `Evaluating` and
+   * `NullValue` are represented by `object`s and `Waiting` by a class that allows awaiting the completion
+   * of the evaluation. Note that since tail-recursive functions are transformed *before* lazy-vals,
+   * this implementation does involve explicit while loop. `PatternMatcher` is coming before `LazyVals`,
+   * therefore the pattern matching is implemented using if-s.
+   *
+   * ```
+   * private @volatile var _x: AnyRef = null
+   * 
+   * def x: A =
+   *   val result = _x
+   *   if result.isInstanceOf[A] then
+   *     result // possible unboxing applied here
+   *   else if result.eq(NullValue) then
+   *     null // possible unboxing applied here
+   *   else
+   *     return x_compute() // possible unboxing applied here
+   * 
+   * private def x_compute() =
```

Comment:
```suggestion
   * private def x_compute(): AnyRef =
```
so that we don't have to guess?